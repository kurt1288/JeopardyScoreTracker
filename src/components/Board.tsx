import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { RootState } from "../store";
import { changePointSelected } from "../store/reducer";

const Board = () => {
    const dispatch = useDispatch();
    const { round } = useSelector((state: RootState) => state);

    const points = [ [200, 400, 600, 800, 1000], [400, 800, 1200, 1600, 2000], [5000] ];

    return (
        <React.Fragment>
            {points[round - 1].map((point, index) => (
                <button className="points" key={index} type="button" value={point} onClick={() => dispatch(changePointSelected(point))}>{ point }</button>
            ))}
        </React.Fragment>
    )
}

export default Board;