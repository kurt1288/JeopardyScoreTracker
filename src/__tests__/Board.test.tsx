import * as redux from 'react-redux';
import Board from '../components/Board';
import { render } from '@testing-library/react';

const useSelectorMock = jest.spyOn(redux, 'useSelector');
const useDispatchMock = jest.spyOn(redux, 'useDispatch');

beforeEach(() => {
    useSelectorMock.mockClear();
    useDispatchMock.mockClear();
});

test("displays first round points", () => {
    useSelectorMock.mockReturnValue({round:1});

    const board = render(<Board />);

    expect(board).toMatchSnapshot();
});

test("displays second round points", () => {
    useSelectorMock.mockReturnValue({round:2});

    const board = render(<Board />);

    expect(board).toMatchSnapshot();
});

test("displays third round points", () => {
    useSelectorMock.mockReturnValue({round:3});

    const board = render(<Board />);

    expect(board).toMatchSnapshot();
})