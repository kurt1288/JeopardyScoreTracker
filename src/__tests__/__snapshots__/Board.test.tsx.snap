// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays first round points 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="points"
        type="button"
        value="200"
      >
        200
      </button>
      <button
        class="points"
        type="button"
        value="400"
      >
        400
      </button>
      <button
        class="points"
        type="button"
        value="600"
      >
        600
      </button>
      <button
        class="points"
        type="button"
        value="800"
      >
        800
      </button>
      <button
        class="points"
        type="button"
        value="1000"
      >
        1000
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="points"
      type="button"
      value="200"
    >
      200
    </button>
    <button
      class="points"
      type="button"
      value="400"
    >
      400
    </button>
    <button
      class="points"
      type="button"
      value="600"
    >
      600
    </button>
    <button
      class="points"
      type="button"
      value="800"
    >
      800
    </button>
    <button
      class="points"
      type="button"
      value="1000"
    >
      1000
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`displays second round points 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="points"
        type="button"
        value="400"
      >
        400
      </button>
      <button
        class="points"
        type="button"
        value="800"
      >
        800
      </button>
      <button
        class="points"
        type="button"
        value="1200"
      >
        1200
      </button>
      <button
        class="points"
        type="button"
        value="1600"
      >
        1600
      </button>
      <button
        class="points"
        type="button"
        value="2000"
      >
        2000
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="points"
      type="button"
      value="400"
    >
      400
    </button>
    <button
      class="points"
      type="button"
      value="800"
    >
      800
    </button>
    <button
      class="points"
      type="button"
      value="1200"
    >
      1200
    </button>
    <button
      class="points"
      type="button"
      value="1600"
    >
      1600
    </button>
    <button
      class="points"
      type="button"
      value="2000"
    >
      2000
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`displays third round points 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="points"
        type="button"
        value="5000"
      >
        5000
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="points"
      type="button"
      value="5000"
    >
      5000
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
